tests/roles/backend_services/tasks/main.yaml

```diff
@@ -32,116 +32,43 @@
         oc set data secret/osp-secret "PlacementPassword={{ placement_password }}"
     {% endif %}
 
-- name: deploy backend services
+- name: Gather min facts if we dont have ansible_user_dir
+  ansible.builtin.setup:
+    gather_subset: "!min,user_dir"
+  when: ansible_user_dir is not defined
+
+- name: when not periodic, just run kustomize on the base/deployment
+  when: not periodic|default(false)
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc apply -f - <<EOF
-    apiVersion: core.openstack.org/v1beta1
-    kind: OpenStackControlPlane
-    metadata:
-      name: openstack
-    spec:
-      secret: osp-secret
-      storageClass: local-storage
-
-      cinder:
-        enabled: false
-        template:
-          cinderAPI: {}
-          cinderScheduler: {}
-          cinderBackup: {}
-          cinderVolumes: {}
-
-      dns:
-        enabled: true
-        template:
-          externalEndpoints:
-          - ipAddressPool: ctlplane
-            loadBalancerIPs:
-            - 192.168.122.80
-          options:
-          - key: server
-            values:
-            - 192.168.122.1
-          replicas: 1
-
-      glance:
-        enabled: false
-        template:
-          glanceAPIInternal: {}
-          glanceAPIExternal: {}
-
-      horizon:
-        enabled: false
-        template: {}
-
-      ironic:
-        enabled: false
-        template:
-          ironicConductors: []
-
-      keystone:
-        enabled: false
-        template: {}
-
-      manila:
-        enabled: false
-        template:
-          manilaAPI: {}
-          manilaScheduler: {}
-          manilaShares: {}
-
-      mariadb:
-        templates:
-          openstack:
-            storageRequest: 500M
-
-      memcached:
-        enabled: true
-
-      neutron:
-        enabled: false
-        template: {}
-
-      nova:
-        enabled: false
-        template: {}
-
-      ovn:
-        enabled: false
-        template:
-          ovnController:
-            external-ids:
-              system-id: "random"
-              ovn-bridge: "br-int"
-              ovn-encap-type: "geneve"
-
-      placement:
-        enabled: false
-        template: {}
-
-      rabbitmq:
-        templates:
-          rabbitmq:
-            externalEndpoint:
-              loadBalancerIPs:
-              - 172.17.0.85
-              ipAddressPool: internalapi
-              sharedIP: false
-            replicas: 1
-          rabbitmq-cell1:
-            externalEndpoint:
-              loadBalancerIPs:
-              - 172.17.0.86
-              ipAddressPool: internalapi
-              sharedIP: false
-            replicas: 1
-
-      telemetry:
-        enabled: false
-        template: {}
-    EOF
+    kustomize build base > {{ ansible_user_dir }}/test_deployment.yaml
+    oc apply -f {{ ansible_user_dir }}/test_deployment.yaml
+  args:
+    chdir: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/config"
```

Comment:
This path looks CI-specific? It would be good to figure out a way to use a path relative to the DPA repo checkout, not assuming anything about where the DPA repo itself resides.

Reply:
ack updating... i am going to try a relative path ... we should be in the data-plane-adoption directory via: https://github.com/rdo-infra/rdo-jobs/blob/b6da50ff2d484224a11b0860bf6f21bd7733aab2/playbooks/data_plane_adoption/run_repo_tests.yaml#L144-L146